### PR TITLE
I18n: Translate phrases for new Browse Dashboards

### LINKED ISSUE #70085: i18n for managing dashboards
As we've updated the UI for managing folders, there are strings that need to be marked up for translation, and not hard coded. This is required for the GA release.

Folder view:
image

Move/Delete dialogs:
image
image
(note: for the word they need to type, I expect it needs to stay "delete"

New folder drawer:
image

### PR DESCRIPTION
**Pull request by Bohdanator**
**What is this feature?**

This PR marks strings in the new folder management UI for translation. 

**Which issue(s) does this PR fix?**:

Fixes #70085 

**Special notes for your reviewer:**

I am completely new to this project :). I was able to set up Grafana locally, but I wasn't able to toggle `nestedFolders` feature. After putting it in `custom.ini` config file nothing happened (also after restart). What else is required?

I would also like feedback on the translation IDs. This is my first time working with `i18n` or Crowdin, so I'm not sure what else is required for these translation IDs to be added. (Also if the ID structure makes sense.) 

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by joshhunt** [comment]
We'll check this more in the week, but to enable a feature flag your grafana server config (custom.ini) should look like this

```
[feature_toggles]
nestedFolders=true
```

Also be sure to run `yarn i18n:extract` after marking up any new phrases.

File: public/app/features/browse-dashboards/components/BrowseActions/DeleteModal.tsx [hunk 1]
```diff
@@ -17,6 +18,13 @@ export interface Props {
 
 export const DeleteModal = ({ onConfirm, onDismiss, selectedItems, ...props }: Props) => {
   const [isDeleting, setIsDeleting] = useState(false);
+  const confirmButtonText = useMemo(
+    () =>
+      isDeleting
+        ? t('browse-dashboards.action.deleting', 'Deleting...')
+        : t('browse-dashboards.action.delete-button', 'Delete'),
+    [isDeleting]
+  );
```
Comment (RoxanaAnamariaTurc) [inline public/app/features/browse-dashboards/components/BrowseActions/DeleteModal.tsx:27]: I don't think the useMemo hook is necessary in this case, you can just use the t() function inside the ternary directly

File: public/app/features/browse-dashboards/components/BrowseActions/MoveModal.tsx [hunk 2]
```diff
@@ -19,6 +20,11 @@ export interface Props {
 export const MoveModal = ({ onConfirm, onDismiss, selectedItems, ...props }: Props) => {
   const [moveTarget, setMoveTarget] = useState<string>();
   const [isMoving, setIsMoving] = useState(false);
+  const moveButtonText = useMemo(
+    () =>
+      isMoving ? t('browse-dashboards.action.moving', 'Moving...') : t('browse-dashboards.action.move-button', 'Move'),
+    [isMoving]
+  );
```
Comment (RoxanaAnamariaTurc) [inline public/app/features/browse-dashboards/components/BrowseActions/MoveModal.tsx:27]: See above comment about useMemo hook.

**Review by RoxanaAnamariaTurc — COMMENTED** [review]
@Bohdanator thank you for your contribution, please check the comments I left for you.

**Review by RoxanaAnamariaTurc — APPROVED** [review]
Hi, we merged main and resolved some merge conflicts , we had some phrases that were already translated and we went with what was already in main. Thank you for contributing.

**Pull request merged**